visualization/app/codeCharta/util/mapBuilder.ts

```diff
@@ -1,53 +1,35 @@
-import { CodeMapNode, NodeType } from "../codeCharta.model"
+import { CodeMapNode } from "../codeCharta.model"
 import { CodeChartaService } from "../codeCharta.service"
 
 export class MapBuilder {
+	// TODO: Check if it's fine to keep this unordered.
 	static createCodeMapFromHashMap(hashMapWithAllNodes: Map<string, CodeMapNode>) {
-		let rootNode = this.getEmptyRootNode()
-		const sortedNodes = this.getHashMapSortedByPath(hashMapWithAllNodes)
-
-		sortedNodes.forEach((node: CodeMapNode, path: string) => {
+		let rootNode: CodeMapNode
+		for (const [path, node] of hashMapWithAllNodes) {
 			node.children = []
-			const parentNode = this.getParentNode(sortedNodes, path, rootNode)
+			const parentNode = this.getParentNode(hashMapWithAllNodes, path)
 			if (node.path === CodeChartaService.ROOT_PATH) {
 				rootNode = node
 			} else {
 				parentNode.children.push(node)
 			}
-		})
-		return rootNode
-	}
-
-	private static getEmptyRootNode(): CodeMapNode {
-		return {
-			name: CodeChartaService.ROOT_NAME,
-			path: CodeChartaService.ROOT_PATH,
-			type: NodeType.FOLDER,
-			children: [],
-			attributes: {}
 		}
+		return rootNode
 	}
 
-	private static getHashMapSortedByPath(hashMap: Map<string, CodeMapNode>) {
-		return new Map([...hashMap.entries()].sort((a, b) => a[0].length - b[0].length))
-	}
-
-	private static getParentPath(path: string) {
-		return path.slice(0, Math.max(0, path.lastIndexOf("/")))
-	}
-
-	private static getParentNode(sortedHashMap: Map<string, CodeMapNode>, path: string, rootNode: CodeMapNode): CodeMapNode {
+	private static getParentNode(hashMap: Map<string, CodeMapNode>, path: string): CodeMapNode {
```

Comment:
return type can be undefined as well
```suggestion
	private static getParentNode(hashMap: Map<string, CodeMapNode>, path: string): CodeMapNode | undefined {
```

should this be declared?

Reply:
We have a lot of functions that may return undefined. Currently TypeScript is pretty lax when it comes to these and does not check for undefined. Thus, I did not explicitly add it, while it's definitely correct (strict null checks would be the one to complain).